src/app/ApplicationTemplate.Business/KillSwitch/KillSwitchService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Reactive.Linq;
+using ApplicationTemplate.DataAccess;
+using Microsoft.Extensions.Logging;
+
+namespace ApplicationTemplate.Business;
+
+/// <summary>
+/// Implementation of the IKillSwitchService.
+/// </summary>
+public sealed class KillSwitchService : IKillSwitchService, IDisposable
+{
+	private readonly IKillSwitchRepository _killSwitchRepository;
+	private readonly ILogger<KillSwitchService> _logger;
+	private readonly IDisposable _subscription;
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="KillSwitchService"/> class.
+	/// </summary>
+	/// <param name="killSwitchRepository">The <see cref="IKillSwitchRepository"/>.</param>
+	/// <param name="logger">The <see cref="ILogger{KillSwitchService}"/>.</param>
+	public KillSwitchService(IKillSwitchRepository killSwitchRepository, ILogger<KillSwitchService> logger)
+	{
+		_killSwitchRepository = killSwitchRepository;
+		_logger = logger;
+
+		_subscription = _killSwitchRepository.ObserveKillSwitchActivation()
+			.Subscribe(isActive =>
```

Comment:
Instead of creating another subscription, you could simply implement the log in the `ObserveKillSwitchActivation` member, using a `.Do(isActive => _logger.LogInformation(...))`.